### PR TITLE
Remove msg_snoop MAVLink functionality

### PR DESCRIPTION
**Pull request by peterbarker**
We can now use inheritance to achieve the same result of this but in a cleaner fashion.

### CODE REVIEW

**Comment by magicrub** [comment]
If I want the UAVCAN to fwf all MAVLink packets, is this a similar method I should use?

**Comment by peterbarker** [comment]
@magicrub Yes.

A few notes:
 - this makes use of a singleton.   If you don't have a uavcan singleton then the vehicles will need to implement a callback to return their instance variable.
 - be wary of performance.  If you're forwarding a packed mavlink_message that's not a problem.  If you're decoding everything before reencoding it or something that would be a problem...
 - you may be able to hook into `handleMessage` rather than `packetReceived`; `handleMessage` will only have packets received that this aircraft should handle, `packetReceived` is all of them.

File: AntennaTracker/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -393,33 +394,34 @@ void Tracker::mavlink_snoop(const mavlink_message_t* msg)
...
         // decode
         mavlink_scaled_pressure_t packet;
-        mavlink_msg_scaled_pressure_decode(msg, &packet);
-        tracking_update_pressure(packet);
+        mavlink_msg_scaled_pressure_decode(&msg, &packet);
+        tracker.tracking_update_pressure(packet);
         break;
     }
     }
+    GCS_MAVLINK::packetReceived(status, msg);
```
Comment (lucasdemarchi) [inline AntennaTracker/GCS_Mavlink.cpp]: shouldn't you have a `bool handled` and only call the parent if the packet was **not** handled?

Comment (peterbarker) [reply AntennaTracker/GCS_Mavlink.cpp]: I don't think so.  That would preclude the parent doing other funky things with the received packet - most notable updating the routing tables.

**Comment by peterbarker** [comment]
This is only WIP because I have yet to test it on real vehicles.

File: AntennaTracker/GCS_Mavlink.cpp (hunk 1 above)
Comment (lucasdemarchi) [reply AntennaTracker/GCS_Mavlink.cpp]: couldn't that be done by whoever call this function?  then this follows the normal flow of having a default shared handled and allow specializations to do their own if they need. E.g. https://gist.github.com/fdf971c5e377be183fe2babfe12c2b06

**Comment by peterbarker** [comment]
On Tue, 20 Mar 2018, Lucas De Marchi wrote:

> couldn't that be done by whoever call this function? then this follows the
> normal flow of having a default shared handled and allow specializations to
> do their own if they need. E.g.
> https://gist.github.com/fdf971c5e377be183fe2babfe12c2b06

It could, but packetReceived was actually pulled out of there to try to 
make that block legible (and to allow ADSB to grab all packets IIRC)

**Comment by lucasdemarchi** [comment]
> It could, but packetReceived was actually pulled out of there to try to
> make that block legible (and to allow ADSB to grab all packets IIRC)
ok

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**